packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -660,6 +660,11 @@ class _UiKitViewState extends State<UiKitView> {
   }
 
   void _onFocusChange(bool isFocused) {
+    assert(_controller != null, 'warning: attempt to focus a UIKitView when it is not ready. ');
```

Comment:
Sorry, I said "warning" but it shouldn't be a warning. We should just say something like:
'_controller must be initialized before _focusNode'

Reply:
Or better, let the compiler enforce it.  Something like
```dart
  Widget build(BuildContext context) {
    final UiKitViewController? controller = _controller;
    if (controller == null) {
      return const SizedBox.expand();
    }
    return Focus(
      focusNode: _focusNode,
      onFocusChange: (bool isFocused) => _onFocusChange(isFocused, controller),
      child: _UiKitPlatformView(
        controller: controller,
        hitTestBehavior: widget.hitTestBehavior,
        gestureRecognizers: widget.gestureRecognizers ?? _emptyRecognizersSet,
      ),
    );
  }
```
```dart
  void _onFocusChange(bool isFocused, UiKitViewController controller) {
    if (!isFocused) {
      // Unlike Android, we do not need to send "clearFocus" channel message
      // to the engine, because focusing on another view will automatically
      // cancel the focus on the previously focused platform view.
      return;
    }
    SystemChannels.textInput.invokeMethod<void>(
      'TextInput.setPlatformViewClient',
      <String, dynamic>{'platformViewId': controller.id},
    );
  }
```

Reply:
nice!